### ISSUE TITLE
Add storage module for JVM

### ISSUE THREAD

**Issue by jakubkinst** [GitLiveApp/firebase-java-sdk#4]
I would really appreciate storage module implementation for the JVM target. Is there any particular blocker for this one? Thanks

**Comment by CRamsan** [comment]
I can help with this. :+1:  Do you guys have a slack/discord for some quick questions? I wanted to run some questions before getting started.

**Comment by CRamsan** [comment]
My goal is to be able to use Firebase Storage for a Compose Desktop app. So this would be my approach. 
- As described [here](https://github.com/GitLiveApp/firebase-java-sdk.git), I need to use the Android SDK. This SDK will be exposed as a JVM target with the help of Robolectric to provide any functionality that require Android-specific APIs.
  - On a side-note, is this the long term plan?
- Given that this set up is a bit challenging, I will need to add several unit tests to validate the functionality.
- Once the java-sdk has been updated with JVM support for Storage, I will go to https://github.com/GitLiveApp/firebase-kotlin-sdk/blob/master/firebase-storage/src/jvmMain/kotlin/dev/gitlive/firebase/storage/storage.jvm.kt and wire up the JVM implementation.
- Add some more tests.
- Anything else?

**Comment by nbransby** [comment]
No that's exactly it 👍 you can post any issues you encounter here

**Comment by CRamsan** [comment]
I see that in the [build.gradle.kts](https://github.com/GitLiveApp/firebase-java-sdk/blob/ac9573078603259f05b701606f6218cc834d1d5f/build.gradle.kts), we are exposing all the classes from all the jars. 

```
 api(fileTree(mapOf("dir" to "build/jar", "include" to listOf("*.jar"), "exclude" to listOf("lifecycle-*"))))
```

Is this required or can we have it just use `implementation`? Otherwise, all the jars will be added to the classpath and can very easily cause runtime exceptions even for components that are not firebase related.  

For example, today I spent an hour investigating a crash in my navigation code. The crash was due to a missing method inside the JB Navigation  due to the `firebase-java-sdk` exposing it's own version of several `androidx.*` dependencies.

**Comment by CRamsan** [comment]
This is what I have so far #18 . I still need to do some more testing.

**Comment by nbransby** [comment]
Yes it's required because the library consumers client code compiles and run directly against these android libraries, the code inside this project is an implementation of the android system using the standard java libraries required by the firebase libraries.

**Comment by CRamsan** [comment]
I dont think we need all those classes though. Maybe it would be better to make it an include list for only the classes or packages that are required. Although I dont know how many those are :thinking: . 
Otherwise, it will be a problem now that a lot of the AndroidX libraries are becoming multi-platform, and there will be classes that are `androidx` which will be used in the JVM target as part of those libraries.

**Comment by nbransby** [comment]
Any library excluded from the jars will need to be added via a gradle implementation cause so the code will still compile and run

**Comment by CRamsan** [comment]
My priorities had to change so I will not be needing the storage SDK anymore. Here are the changes I had started, the library loads but I am getting a 400 response when trying to download any file.

Java-sdk changes: #18 
Kotlin-sdk changes: [branch](https://github.com/CRamsan/firebase-kotlin-sdk/tree/cramsan/storage_jvm)

**Comment by pontomedon** [comment]
@CRamsan I updated your branches to latest `master` and finished the work. Seems to be all working!

- firebase-java-sdk: https://github.com/GitLiveApp/firebase-java-sdk/pull/44
- firebase-kotlin-sdk: https://github.com/GitLiveApp/firebase-kotlin-sdk/pull/704

**Comment by pontomedon** [comment]
@nbransby is the firebase sdk still something that's being developed? I'd be happy for a PR review :)
